addons/test_mail/tests/test_message_post.py

```diff
@@ -239,16 +239,21 @@ def test_post_notify(self):
         new_notification = self.test_record.message_notify(
             subject='This should be a subject',
             body='<p>You have received a notification</p>',
-            partner_ids=[self.partner_1.id, self.user_employee.partner_id.id],
+            partner_ids=[self.partner_1.id, self.partner_admin.id, self.user_employee.partner_id.id],
         )
 
         self.assertEqual(new_notification.subtype_id, self.env.ref('mail.mt_note'))
         self.assertEqual(new_notification.message_type, 'user_notification')
         self.assertEqual(new_notification.body, '<p>You have received a notification</p>')
         self.assertEqual(new_notification.author_id, self.env.user.partner_id)
         self.assertEqual(new_notification.email_from, formataddr((self.env.user.name, self.env.user.email)))
-        self.assertEqual(new_notification.notified_partner_ids, self.partner_1 | self.user_employee.partner_id)
+        self.assertEqual(new_notification.notified_partner_ids, self.partner_1 | self.user_employee.partner_id | self.partner_admin)
         self.assertNotIn(new_notification, self.test_record.message_ids)
+
+        body = self._mails[0].get("body")
```

Comment:
I fear this may break depending on order to sent emails. Indeed normally you should have 2 outgoing emails: 1 for partner_1, 1 for partner_admin. Could be great to check that you have 2 outgoing emails, 1 without link, 1 with link. You can base yourself on email_to of _mails to find the right one (partner or admin).